code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -49,7 +49,6 @@
 	R.speed = 0 // Remove upgrades.
 	R.ionpulse = FALSE
 	R.magpulse = FALSE
-	R.weather_immunities = initial(R.weather_immunities)

```

Comment:
it's not broken, it just doesn't do lists.
rather it doesn't do objects, and a list is an object.